### ISSUE TITLE
Paper code reproduction help

### ISSUE THREAD

**Issue by xiaoli595** [aymane399/AFSC#2]
Dear author,
 I am very interested in the algorithmic framework of your paper and am committed to doing research based on it. But I followed the steps you provided to download the pre-trained features of CUB trained by resnet12, and selected LSS for training as a standard. The final weighted accuracy is far from the results you obtained. I would like to ask about your environment configuration and whether there is any Latest code version?
Thank you for answering my question.Have a nice day!

**Comment by aymane399** [comment]
Hello Xiaoli,

Thank you for your interest in my research, I will check and get back to
you today.
 Have a great day,

Aymane

**Comment by xiaoli595** [comment]
![Snipaste_2024-04-17_17-02-47](https://github.com/aymane399/AFSC/assets/107289643/89a2065a-9260-4ce2-bd00-46a2754ec973)
![Snipaste_2024-04-17_18-44-08](https://github.com/aymane399/AFSC/assets/107289643/6bc12834-89fe-4a3c-9651-7ff7edf886d6)

Dear Aymane,
I am using the experiment run under torch1.9.1+cuda11.1 configuration. No matter what criteria I run in the experiment, such as andom, margin, etc., the weighted accuracy with n_labels of 5 is very different from your results, while the performance with n_labels of 25 is not much different. I'm a little confused and wondering what went wrong.
Thanks for your help, it's greatly appreciated.
Lee

**Comment by aymane399** [comment]
Hello Lee,

Thank you for reporting this, there was a bug in the sampler that is now fixed,  just pull the latest commit. Please feel free to reach out if you notice any other issues.

Aymane

**Comment by xiaoli595** [comment]
Dear Aymane,
Thank you for your work. The download link for the Google Drive features of TIM resnet 12 on your page is invalid. I wonder if you can take the time to update it. Thanks again for all your hard work!
Lee

**Comment by aymane399** [comment]
You're welcome :) 

The original paper who produced the TIM features (https://github.com/mboudiaf/TIM) changes the hosting link to iCloud : you should find them here : https://www.icloud.com/iclouddrive/0f3PFO3rJK0fk0nkCe8sDKAiQ#TIM.
